model/atmosphere/dycore/README.md

```diff
@@ -4,6 +4,66 @@
 
 Contains code ported from ICON `src/atm_dyn_iconam`, which is the dynamical core of the ICON model.
 
+## Attention
+
+Variables in ICON can either sit on full levels or half levels. Let's denote the number of full levels is nlev.
+Full-level variables have nlev levels in vertical dimension, while half-level variables have nlev+1 in vertical dimension.
+Do not output full-level and half-level variables together when running a gt4py program if you need to access the nlev+1 level because it will cause random behavior (most probably illegal memory access).
```

Comment:
Could you add a short explanation of *why* the random behaviour/out of bounds accesses are caused if we have a stencil where the output fields are half level and full level kdim variables, this isn't clear from the description and might be useful for the reader to understand.

Reply:
Yes, it is important here to differentiate between a program and field_operator. If a `field_operator has 2 variables with different output domains (nlev and nlevp1) there will be a problem. However, you can do this in a program, by calling 2 different field_operators for the two variables with different domains, that will work.

Reply:
Thanks for the comment. I have modified to emphasize the point where the user cannot output 2 variables with different vertical domains in a `field_operator`.